.vscode/settings.json

```diff
@@ -11,26 +11,27 @@
   "autoDocstring.guessTypes": true,
   "autoDocstring.includeExtendedSummary": false,
   "autoDocstring.generateDocstringOnEnter": false,
-  "python.formatting.provider": "black",
-  "python.sortImports.path": "isort",
-  "python.linting.enabled": true,
-  "python.linting.lintOnSave": true,
-  "python.linting.pylintEnabled": true,
-  "python.linting.pylintArgs": [
+  "pylint.args": [
     "--rcfile",
     "./lib/pymedphys/.pylintrc"
   ],
-  "python.linting.mypyEnabled": true,
-  "python.linting.pylintUseMinimalCheckers": false,
   "[python]": {
     "editor.rulers": [
       72,
       88
-    ]
+    ],
+    "editor.formatOnSave": true,
+    "editor.codeActionsOnSave": {
+      "source.fixAll": "explicit",
+      "source.organizeImports": "explicit"
+    },
+    "editor.defaultFormatter": "charliermarsh.ruff",
+  },
+  "notebook.formatOnSave.enabled": true,
```

Comment:
Even though it's supported, I've struggled with `black` for Jupyter notebooks, at least when autoformatting in VS Code. It sometimes adds excessive whitespace. Have you tried `ruff` specifically for this?

Reply:
I haven't, but I've heard good things. We can remove if it causes issues.